src/checkers/inference/InferenceMain.java

```diff
@@ -142,6 +143,13 @@ public void run() {
         // Start up javac
         startCheckerFramework();
         solve();
+        // solverResult = null covers case when debug solver is used, but in this case
+        // shouldn't exit
+        if (solverResult != null && !solverResult.hasSolution()) {
+            // TODO Do something to stop the inferene process
+            logger.info("No solution, exiting...");
+            System.exit(1);
+        }
```

Comment:
Exit the inference process if there is no solution. This fixes problem of printing "inference succeeded" even though there is no solution.
This might be not the best place to add the logic. But due to the fact that CFI has a very complicated launcher classes, let's at least improve it a little bit.